cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -112,6 +116,25 @@ public final ProgramController run(final Program program, final ProgramOptions o
       throw Throwables.propagate(e);
     }
 
+    URI logbackURI = null;
+    try {
+      // TODO: When CDAP-1273 is fixed you can get the resource directly from the program classloader.
+      // Make an unused call to getClassloader() to ensure that the jar is expanded into programDir.
+      copiedProgram.getClassLoader();
+      File logbackFile = new File(programDir, "logback.xml");
```

Comment:
Thats what I originally did, but because of https://issues.cask.co/browse/CDAP-1273 it wasn't as clean. Essentially the `getResource` method for `Program`s will get the parent resource before the program resource.